smartapps/rachio/rachio-connect.src/rachio-connect.groovy

```diff
@@ -1281,11 +1282,15 @@ def getControlLblById(id) {
 }
 
 def getCurrentRainDelay(res) {
-	//log.debug("getCurrentRainDelay($devId)...")
-	// convert to configured rain delay to days.
-	def ret =  (res?.rainDelayExpirationDate || res?.rainDelayStartDate) ? (res?.rainDelayExpirationDate - res?.rainDelayStartDate)/(26*60*60*1000) : 0
-	def value = (long) Math.floor(ret + 0.5d)
-	return value
+    //log.debug("getCurrentRainDelay($devId)...")
+    // convert to configured rain delay to days.
+    // as it seems rainDelayStartDate is never returned in the API response and it seems the value doesn't change using hardcoded default value of
+    // 1534193839000 as (1534287439000 - 1534193839000)/(26*60*60*1000) = 1
+    //def ret =  (res?.rainDelayExpirationDate || res?.rainDelayStartDate) ? (res?.rainDelayExpirationDate - res?.rainDelayStartDate)/(26*60*60*1000) : 0
+    def rainDelayStartDate = 1534193839000
```

Comment:
`def rainDelayStartDate = res?.rainDelayStartDate ?: 1534193839000`

Reply:
set rainDelayStartDate to current date